### ISSUE TITLE
Notifier is not present on AuthorizationRequest handler. No delivery of result will be possible

### ISSUE THREAD

**Issue by weihsth** [wi3land/ionic-appauth#323]
Hi folks,
thank you for this great plugin and the effort of adding examples!

I already searched through the open issues and googled a lot, but can not find any helping advices for my bug(?).

Using the capacitor example code the project compiles without errors. Login and redirect (locally) is working well until it comes to calling auth.authorizationCallback(). This always ends up in a dev-tools-termin message like this:
```
logger.js:34 Checking to see if there is an authorization response to be delivered.
logger.js:34 Notifier is not present on AuthorizationRequest handler. No delivery of result will be possible
```

My authConfig (KeyCloak) looks like this:
```
const sharedAuthConfig = {
    client_id: 'web',
    server_host: 'https://auth.my-domain.com/auth/realms/<realm-name>',
    redirect_url: '<app-ident>://auth/callback',
    end_session_redirect_url: '<app-ident>://auth/end-session',
    scopes: 'openid profile email roles offline_access',
    pkce: true, // tested with true and false
};
```

Client is public and was working with another oidc plugin I used before. I switched to yours because of ionic app support.

auth.factory.ts:
```
    [...]

    if (!platform.is('cordova')) {
        authService.authConfig.redirect_url = window.location.origin + '/auth/callback';
        authService.authConfig.end_session_redirect_url = window.location.origin + '/auth/end-session';
    }

    [...]
```

The '/auth/callback' is reached, bot nothing happens on this page. It shoul redirect me to "/home" after checking the auth code. I dont even see any "/token" request in the dev tools, so the auth code is not handled.

I see the callback-URL with all the parameters in the Browser address field. When refreshing the browser, I can see this error event:
```
{ "action": "Sign In Failed", "error": "Handle Not Available" }
```

The auth.init() is called in AppComponent constructor after platform.ready(). Also tried to run the init before platform.ready(). Does not matter. From the source code I see, that the notifier should be set up when calling init() funtion. But seems, it does not.

Since I cannot find any advices in documentation about which kind of Notifier or AuthorizationRequest handler is ment, I ask you kindly for giving me any advice, link or example, what I need to adjust.

I checked code a lot of times and even went throgh cordova and other example code. There is nowwhere a setup für this kind of classes.

Regards!

**Addendum:**
Tested with Android (deployed via Android Studio). Redirect to login works. Redirect back to /auth/callback works, but from the $events observable I get this error:
```
{ "action": "Sign In Failed", "error": "Handle Not Available" }
```

I looked in the capacitor example code again and can not find any "handle" setup.

**Comment by weihsth** [comment]
After checking the lib code, I found out, that I am not allowed to set the response_type. For example for lokal development I am always using `response_type: 'token'`, which is also supported by AppAuth-JS. Is there a reason, why this part is hard codet in the module lib source code? Do you plan to change this?

```performAuthorizationRequest(authExtras, state) {
        return __awaiter(this, void 0, void 0, function* () {
            let requestJson = {
                response_type: AuthorizationRequest.RESPONSE_TYPE_CODE, // <--- not possible to configure :(
                client_id: this.authConfig.client_id,
                redirect_uri: this.authConfig.redirect_url,
                scope: this.authConfig.scopes,
                extras: authExtras,
                state: state || undefined,
            };
            let request = new AuthorizationRequest(requestJson, new DefaultCrypto(), this.authConfig.pkce);
            if (this.authConfig.pkce)
                yield request.setupCodeVerifier();
            return this.requestHandler.performAuthorizationRequest(yield this.configuration, request);
        });
    }```

**Comment by weihsth** [comment]
Hello folks,

is there really noone with any hint why I get errors on the callback page?

Redirect to login-page (keycloak) works, redirekt to callback page (app) works.

But on the callback page I get the error: {"action":"Sign In Failed","error":"Handle Not Available"}

I just need a hint, what I need to install, activate or configure to get this magic "Handle" available. There is no hint in the documentation about this. And I searched through all open and closed Issues in this project without any success.

**Comment by dmoojunk** [comment]
Hey, just had this myself. This notifier is set on auth.init(). After a redirect you need to make sure this has been called before it reaches your callback page. My problem was I was trying to isolate this in a test area and did not have the auth init happening early enough in the sequence. I needed to move the auth init to the app component constructor so notifier existed.

**Comment by weihsth** [comment]
I am doing the init() call right after platform.ready(). So that was not the issue..

But I fixed it today. The real Issue was, that I added AuthService to "prviders: []" list in two different modules. So the init() call was done on the first instance, and the rest on the second instance.

It is working now! I added AuthService to AppModule only now. And the imported AuthModule is using this instance.

Anyway thank xou for your response @dmoojunk !
